packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -61,39 +89,153 @@ contract FeeHandler is Ownable, Initializable, UsingRegistry, ICeloVersionedCont
 
   /**
    * @notice Used in place of the constructor to allow the contract to be upgradable via proxy.
-   * @param _registryAddress The address of the registry core smart contract.
-   * @param tokens A list of tokens whose parameters should be set.
-   * @param newLimits A list of daily burn limits, corresponding with the same order as in the 
-      argument tokens.
-   * @param newMaxSlippages A list of max acceptable slippage, corresponding with the same order as 
-      in the argument tokens.
-   * @param newRouters A list of routers, corresponding with the same order as in the argument 
-      tokens.
    */
   function initialize(
     address _registryAddress,
+    address newFeeBeneficiary,
     address[] calldata tokens,
+    address[] calldata handlers,
     uint256[] calldata newLimits,
-    uint256[] calldata newMaxSlippages,
-    address[] calldata newRouters
+    uint256[] calldata newMaxSlippages
   ) external initializer {
-    require(tokens.length == newLimits.length, "limits length should match tokens'");
-    require(tokens.length == newMaxSlippages.length, "maxSlippage length should match tokens'");
-    require(tokens.length == newRouters.length, "maxSlippage length should match tokens'");
+    require(tokens.length == handlers.length, "limits length should match tokens");
+    require(tokens.length == newLimits.length, "maxSlippage length should match tokens");
+    require(tokens.length == newMaxSlippages.length, "maxSlippage length should match tokens");
 
     _transferOwnership(msg.sender);
     setRegistry(_registryAddress);
+    _setFeeBeneficiary(newFeeBeneficiary);
 
     for (uint256 i = 0; i < tokens.length; i++) {
+      _addToken(tokens[i], handlers[i]);
       _setDailyBurnLimit(tokens[i], newLimits[i]);
       _setMaxSplippage(tokens[i], newMaxSlippages[i]);
-      // Mento tokens don't need to set a router
-      if (newRouters[i] != address(0)) {
-        _setRouter(tokens[i], newRouters[i]);
-      }
     }
   }
 
+  function getTokenHandler(address tokenAddress) external view returns (address) {
```

Comment:
Can we add Natspec to all external functions ?

Reply:
I'll do that when merging the ultragreenbranch